docs/DeveloperGuide.md

```diff
@@ -36,3 +33,470 @@
+    }
+
+    @Override
+    public void printGoalAfterShoot(boolean goalScored, int direction) {
+        Formatter.printGoalAfterShotMedium(goalScored, direction);
+    }
+}
+```
+## PlayerList Class
+
+### Attributes
+
+- `l1`: ArrayList<Player> - Manages a dynamic list of player objects.
+
+### Methods
+
+- `skillUpgrade(int curPlayer)`: Enhances the skill level of the specified player.
+
+### Utilizing ArrayList
+
+The `PlayerList` class showcases the strategic use of `ArrayList<Player>` for player management, chosen for its dynamic resizing capabilities and efficient access times. This choice aligns with our need to manage a variable number of player objects dynamically.
+
+### Leveraging Polymorphism
+
+Polymorphism is integral to our game, allowing us to abstract player actions in the `Player` class and provide specific implementations in its subclasses. This design pattern simplifies codebase maintenance and enhances gameplay by introducing dynamic behavior based on the player's skill level.
+
+#### ExecuteShoot Method
+
+The `executeShoot` method exemplifies polymorphism in action, adjusting player behavior during shooting based on their skill level.
+
+```java
+public static void executeShoot(String[] readArgumentTokens) {
+  float adjustedDirection = PlayerList.l1.get(Ui.curPlayer).shootDirectionAdjust(selectedDirectionIndex);
+  float adjustedAiDirection = PlayerList.l1.get(Ui.curPlayer).aiDirectionAdjust(Ai.getAiDirection());
+  float adjustedRange = PlayerList.l1.get(Ui.curPlayer).rangeAdjust();
+  //...
+  PlayerList.l1.get(Ui.curPlayer).printGoalAfterShoot(isScoreGoal, Math.round(adjustedDirection));
+}
+```
+### Skill Upgrade Mechanism
+
+The `skillUpgrade` method in the `PlayerList` class is a key feature, allowing players to improve their skills based on game performance. The method checks the player's current skill and match count, upgrading their skill level if certain conditions are met.
+he mechanism operates through a structured process, enhancing the gaming experience by offering a realistic approach to skill progression:
+
+1. **Skill Level Assessment**: The system evaluates a player's current skill level alongside their performance metrics, such as match count and success rate.
+
+2. **Determination of Skill Upgrade Eligibility**: Utilizing predefined criteria, the system determines if a player qualifies for a skill upgrade. These criteria may include metrics like a minimum number of matches played or specific performance thresholds.
+
+3. **Application of Skill Upgrade**: Eligible players will have their skill attribute adjusted to a higher level. This upgrade might unlock new abilities, enhance stats, or offer other in-game advantages.
+
+4. **Feedback to the User**: The system informs the user of the successful skill upgrade through UI messages or other feedback mechanisms, providing immediate recognition of their achievement.
+
+![UML Class Diagram](diagrams%2FupdateSkill.png)
```

Comment:
Well documented sequence diagram. Clear presentation of interactions between classes